standalone-metastore/metastore-server/src/test/java/org/apache/hadoop/hive/metastore/VerifyingObjectStore.java

```diff
@@ -93,9 +93,11 @@ public boolean getPartitionsByExpr(String catName, String dbName, String tblName
   @Override
   public List<Partition> getPartitions(
       String catName, String dbName, String tableName, int maxParts) throws MetaException, NoSuchObjectException {
+    openTransaction();
     List<Partition> sqlResults = getPartitionsInternal(catName, dbName, tableName, maxParts, true, false);
     List<Partition> ormResults = getPartitionsInternal(catName, dbName, tableName, maxParts, false, true);
     verifyLists(sqlResults, ormResults, Partition.class);
+    commitTransaction();
```

Comment:
what if exception happens when getPartitionsInternal,  the transaction may become unbalanced

Reply:
This is a test class.  If things become unbalanced here it doesn't really matter because the test will fail from Exception anyway.

Reply:
Got it.